src/components/Router/index.js

```diff
@@ -42,10 +42,18 @@ class CrossDeviceMobileRouter extends Component {
 
   setConfig = (actions) => (data) => {
     const {token, steps, documentType, step} = data
-    this.setState({token, steps, step})
+    const crossDeviceClientSteps = this.replaceLastStep(steps)
+    this.setState({token, steps: crossDeviceClientSteps, step})
     actions.setDocumentType(documentType)
   }
 
+  replaceLastStep = (steps) => {
```

Comment:
what happens if the master cross device doesn't have the complete step?

Reply:
Nothing, we don't care. What we care about is displaying the success screen after captures have been successfully uploaded on the cross device client. The cross device client should always display the success screen at the end of the capture flow and should never show the complete component.

Reply:
speculative: does this become much if we fix the steps when we send them? i.e. in MainRouter.mobileConfig() you could strip out complete and add client success. Then CrossDeviceMobileRouter goes back to being dumb

Reply:
Uhm, but that would mean that MainRouter will have to know about the mobile steps... I've moved this logic in the `stepComponentMap.js` as you guys suggested earlier, it looks cleaner to me. What do you think?